File: src/jvis/data/stacks/nodejs-fastify/files/src/services/item.service.ts
import type { Item } from '../domain/entities/item.js';
import { NotFoundError } from '../domain/errors/app-error.js';
import { prisma } from '../infrastructure/prisma-client.js';

export class ItemService {
  async create(data: { name: string; description?: string | null }): Promise<Item> {
    return prisma.item.create({ data });
  }

  async list(): Promise<Item[]> {
    return prisma.item.findMany({ orderBy: { createdAt: 'desc' } });
  }

  async getById(id: string): Promise<Item> {
    const item = await prisma.item.findUnique({ where: { id } });
    if (!item) {
      throw new NotFoundError('Item', id);
    }
    return item;
  }

  async update(id: string, data: { name?: string; description?: string | null }): Promise<Item> {
    await this.getById(id);
    return prisma.item.update({ where: { id }, data });
  }

  async delete(id: string): Promise<void> {
    await this.getById(id);
    await prisma.item.delete({ where: { id } });
  }
}

File: src/jvis/data/stacks/python-flask/files/src/services/item_service.py
"""Item business logic."""

from src.app import db
from src.models import Item


class ItemService:
    """Service layer for Item CRUD operations."""

    @staticmethod
    def create(name: str, description: str = "") -> Item:
        item = Item(name=name, description=description)
        db.session.add(item)
        db.session.commit()
        return item

    @staticmethod
    def get_all() -> list[Item]:
        return Item.query.order_by(Item.created_at.desc()).all()

    @staticmethod
    def get_by_id(item_id: str) -> Item | None:
        return db.session.get(Item, item_id)

    @staticmethod
    def update(item: Item, name: str | None = None, description: str | None = None) -> Item:
        if name is not None:
            item.name = name
        if description is not None:
            item.description = description
        db.session.commit()
        return item

    @staticmethod
    def delete(item: Item) -> None:
        db.session.delete(item)
        db.session.commit()
